### PR TITLE
Add _encode_url to improve readability of url construction

### PR DESCRIPTION
**Pull request by femtotrader**
A partial fix of https://github.com/pydata/pandas-datareader/issues/78

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
Small remark: the `urlencode` from pandas you use, is in pandas only used in `data.py`, so I think it is better to put it here instead of importing it from pandas (and it is just an import from `urllib` actually: https://github.com/pydata/pandas/blob/df23f918cecbaf755cfdfc559a9ae98131fb390b/pandas/io/common.py#L18)

**Comment by femtotrader** [comment]
So you think we should also have to maintain Python 2 and 3 compatibility here ?

**Comment by TomAugspurger** [comment]
Yes.

**Comment by femtotrader** [comment]
That's done. I wonder why not using six (in Pandas also)

**Comment by jreback** [comment]
https://github.com/pydata/pandas/blob/master/pandas/compat/__init__.py

incorporates six here

File: pandas_datareader/utils.py [hunk 1]
```diff
@@ -85,6 +89,16 @@ def _in_chunks(seq, size):
     """
     return (seq[pos:pos + size] for pos in range(0, len(seq), size))
 
+def _encode_url(url, params):
+    """
+    Return encoded url with parameters
+    """
+    s_params = urlencode(params)
+    if len(s_params)!=0:
```
Comment (hayd) [inline pandas_datareader/utils.py]: `if s_params:`

**Comment by hayd** [comment]
A previous idea was to rip out part of pandas.compat into another lib, as some older pandas have in compat is slightly different. That said,  [we're not using many compat features](https://github.com/pydata/pandas-datareader/search?utf8=%E2%9C%93&q=compat) at all so could just add the few we are to util/_util. :/

**Comment by femtotrader** [comment]
I think I will need some help about this PR.

**Comment by hayd** [comment]
to squash

```
git rebase -i head~8 # squash 8 commits
# change all but the first "pick" to "s", :wq
# clean up the commit message, :wq
```

**Comment by scls19fr** [comment]
Some commit shouldn't be here

    pick bb74632 TST: Fix test fred for new number of datapoints.
    pick 8734d5c TST: pramga: no cover where appropriate
    pick 6dea000 CLN: Add NotImplementedError to DataReader
    pick e5b4039 CLN: Unnecessary try: block removed from Options
    pick df405e2 DOC: update link to docs to point to actual docs
    pick d6e10a1 CLN: Split into subpackages
    pick 2044d8a Add _encode_url to improve readability of url construction
    pick a9ce20e TST: Remove extra 'test_get_quote_string'.
    pick 91e5c2d Add _encode_url to improve readability of url construction
    pick b934e26 Add _encode_url to improve readability of url construction
    pick 0f38c70 Update utils.py
    pick 2b072c8 Add urlencode with Python 2/3 compatibility
    pick 8084c3d if s_params

**Comment by femtotrader** [comment]
Some commit shouldn't be here

    pick bb74632 TST: Fix test fred for new number of datapoints.
    pick 8734d5c TST: pramga: no cover where appropriate
    pick 6dea000 CLN: Add NotImplementedError to DataReader
    pick e5b4039 CLN: Unnecessary try: block removed from Options
    pick df405e2 DOC: update link to docs to point to actual docs
    pick d6e10a1 CLN: Split into subpackages
    pick 2044d8a Add _encode_url to improve readability of url construction
    pick a9ce20e TST: Remove extra 'test_get_quote_string'.
    pick 91e5c2d Add _encode_url to improve readability of url construction
    pick b934e26 Add _encode_url to improve readability of url construction
    pick 0f38c70 Update utils.py
    pick 2b072c8 Add urlencode with Python 2/3 compatibility
    pick 8084c3d if s_params

**Comment by femtotrader** [comment]
I did

    pick 2044d8a Add _encode_url to improve readability of url construction
    squash 91e5c2d Add _encode_url to improve readability of url construction
    squash b934e26 Add _encode_url to improve readability of url construction
    squash 0f38c70 Update utils.py
    squash 2b072c8 Add urlencode with Python 2/3 compatibility
    squash 8084c3d if s_params

but it raises

    error: could not apply 2044d8a... Add _encode_url to improve readability of url construction

    When you have resolved this problem, run "git rebase --continue".
    If you prefer to skip this patch, run "git rebase --skip" instead.
    To check out the original branch and stop rebasing, run "git rebase --abort".
    Could not apply 2044d8a29dae7f5bf92a5a4c1eb657d50103a47f... Add _encode_url to improve readability of url construction

**Comment by hayd** [comment]
That's as you have merge commits, tip: don't use `git merge`.

Maybe `git reset head~2` and then recommit.

**Comment by davidastephens** [comment]
Closed via #83

**Comment by femtotrader** [comment]
Thanks David for your help. You are my Git Guru ;-)

**Pull request closed without merging**